embedded-graphics/src/primitives/polyline.rs

```diff
@@ -0,0 +1,358 @@
+    /// ```
+    fn translate_mut(&mut self, by: Point) -> &mut Self {
+        self.translate += by;
+
+        self
+    }
+}
+
+/// An iterator over all pixel positions on the polyline
+#[derive(Copy, Clone, Eq, PartialEq, Ord, PartialOrd, Hash, Debug)]
+pub struct PolylineIterator<'a> {
+    stop: bool,
+    vertices: &'a [Point],
+    translate: Point,
+    segment_iter: ThickLineIterator,
+}
+
+impl<'a> Iterator for PolylineIterator<'a> {
+    type Item = Point;
+
+    fn next(&mut self) -> Option<Self::Item> {
+        if self.stop {
+            return None;
+        }
+
+        if let Some(p) = self.segment_iter.next() {
+            Some(p)
+        } else {
+            let (start, rest) = self.vertices.split_first()?;
+            let end = rest.get(0)?;
+
+            self.vertices = rest;
+
+            self.segment_iter = ThickLineIterator::new(
+                &Line::new(*start + self.translate, *end + self.translate),
+                1,
+            );
+
+            // Skip first point of next line, otherwise we overlap with the previous line
+            Self::next(self);
+
+            Self::next(self)
+        }
+    }
+}
+
+impl<'a> IntoIterator for Polyline<'a> {
+    type Item = Point;
+    type IntoIter = PolylineIterator<'a>;
+
+    fn into_iter(self) -> Self::IntoIter {
+        self.vertices
+            .split_first()
+            .and_then(|(start, rest)| {
+                // Polyline is 2 or more vertices long, return an iterator for it
+                rest.get(0).map(|end| Self::IntoIter {
+                    stop: false,
+                    vertices: rest,
+                    translate: self.translate,
+                    segment_iter:ThickLineIterator::new(
+                        &Line::new(*start + self.translate, *end + self.translate),
+                        1,
+                    ),
+                })
+            })
+            .unwrap_or_else(||
+                // Polyline is less than 2 vertices long. Return a dummy iterator that will short
+                // circuit due to `stop: true`
+                Self::IntoIter {
+                    stop: true,
+                    vertices: &[],
+                    translate: Point::zero(),
+                    segment_iter: ThickLineIterator::new(&Line::new(Point::zero(), Point::zero()), 1),
+                })
+    }
+}
+
+impl<'a, C> IntoIterator for &'a Styled<Polyline<'a>, PrimitiveStyle<C>>
+where
+    C: PixelColor,
+{
+    type Item = Pixel<C>;
+    type IntoIter = StyledPolylineIterator<'a, C>;
+
+    fn into_iter(self) -> Self::IntoIter {
+        StyledPolylineIterator {
+            style: self.style,
+
+            line_iter: self.primitive.into_iter(),
+        }
+    }
+}
+
+/// Pixel iterator for each pixel in the line
+#[derive(Copy, Clone, Eq, PartialEq, Ord, PartialOrd, Hash, Debug)]
+pub struct StyledPolylineIterator<'a, C>
+where
+    C: PixelColor,
+{
+    style: PrimitiveStyle<C>,
```

Comment:
Using `stroke_color` and `stroke_width` instead of the whole `style` saves some memory.